File: db/csharp/SalesManagement.Domain/Models/Department.cs
namespace SalesManagement.Domain.Models
{
    /// <summary>
    /// 部門マスタのEntityクラス
    /// </summary>
    public class Department
    {
        public string DepartmentCode { get; set; } = string.Empty;
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public string DepartmentName { get; set; } = string.Empty;
        public int OrganizationLevel { get; set; }
        public string DepartmentPath { get; set; } = string.Empty;
        public int LowestLevelFlag { get; set; }
        public int SlipInputFlag { get; set; }
        public DateTime CreatedAt { get; set; }
        public string CreatedBy { get; set; } = string.Empty;
        public DateTime UpdatedAt { get; set; }
        public string UpdatedBy { get; set; } = string.Empty;
    }
}

File: db/fsharp/SalesManagement.Domain/Models/Department.fs
namespace SalesManagement.Domain.Models

open System

/// <summary>
/// 部門マスタのEntityレコード型
/// </summary>
[<CLIMutable>]
type Department = {
    DepartmentCode: string
    StartDate: DateTime
    EndDate: DateTime
    DepartmentName: string
    OrganizationLevel: int
    DepartmentPath: string
    LowestLevelFlag: int
    SlipInputFlag: int
    CreatedAt: DateTime
    CreatedBy: string
    UpdatedAt: DateTime
    UpdatedBy: string
}
